### ISSUE TITLE
creatures unable to find a path

### ISSUE THREAD

**Issue by ghost** [Varkalandar/imp_city#2]
![2023-11-04](https://github.com/Varkalandar/imp_city/assets/1316300/66d59cfd-50d3-4ba1-9936-04ee81d1c866)

In this screenshot, I had to dig the long path for snakes to be able to reach the library. With only the simple door, they were somehow unable to find their path to library.

This might be because a library item sits right in front of the entrance.

Building the access led to a cascade of discoveries, too, as if creatures were actually using the library despite not being able to reach it.

**Comment by Varkalandar** [comment]
I see. Thank you for reporting this. Indeed the book shelf blocks the path. I'll look into solutions, maybe it will be sufficient to move the bookshelf a bit more to the center of the tile.

The cascade of discoveries is a different problem, but I think you also found a real bug there. I'll try to reproduce it and look for a fix. Thanks again for the report!

**Comment by ghost** [comment]
> I see. Thank you for reporting this. Indeed the book shelf blocks the path. I'll look into solutions, maybe it will be sufficient to move the bookshelf a bit more to the center of the tile.

IIRC, dungeon keeper would only add items on claimed walls and in middle of an at least 3x3 area. Doing the same would both get rid of the problem of obstacles spawning in front of a door, and of your concerns about players removing all walls, as walls would become more efficient space-wise.
I may be wrong, but I think some room types had bigger requirements to start being usable at all.

> The cascade of discoveries is a different problem, but I think you also found a real bug there. I'll try to reproduce it and look for a fix. Thanks again for the report!

Well, I say this because I have the impression that this also applies in forges: when a monster does it's forge work without any input resource, when imps go drop some resources, they drop bronze or something. Or maybe I missed something, I'm not entirely sure.

**Comment by Varkalandar** [comment]
I've started to change the room furnishing code. So far I've got the libraries done. A 2x2 library will be the first size to receive a shelf (and thus the first size that actually can be used for research)

Forges use a different timing approach than research or quests, there is an counter for work steps, and every 64 steps a creature working in the forge will consume inputs and produce something. If this happens just after an jmp brought new resources, tge production will appear immediately.
